### PR TITLE
IIO demo variable definition update

### PR DESCRIPTION
**Pull request by adrimbarean**
Define global variables used in IIO modules of adc_demo and dac_demo in the source files.
Improve the ability to have variable number of channels.

### CODE REVIEW

File: drivers/adc/adc_demo/iio_adc_demo.c [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+	.ch_type = IIO_VOLTAGE,\
+	.channel = _idx,\
+	.scan_index = _idx,\
+	.indexed = true,\
+	.scan_type = &adc_scan_type,\
+	.attributes = adc_channel_attributes,\
+	.ch_out = false,\
+}
+
+struct iio_channel *iio_adc_demo_channel(uint8_t ch_no)
```
Comment (mchindri) [inline drivers/adc/adc_demo/iio_adc_demo.c:81]: we don't want to do this. This should be as an example for other drivers.
And the best approach for a new driver with a fixed number of channels is to declare the structure.
In order to have a variable number of channels will be easy to set num_ch as the desired number and declare 16 channels.
Something like: 
`
#ifndef DEFAULT_ADC_CH_NUMBER
#define DEFAULT_ADC_CH_NUMBER 2
#endif
struct iio_device iio_adc_demo_dev =  { 
	.num_ch = DEFAULT_ADC_CH_NUMBER, 
	.channels = all_iio_adc_channels, 
....

`

**Comment by adrimbarean** [comment]
V1:
 - use "#define" to define the number of channels implemented in the driver.

**Review by amiclaus — APPROVED** [review]

File: drivers/dac/dac_demo/iio_dac_demo.c [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+	DAC_DEMO_ATTR("dac_channel_attr", DAC_CHANNEL_ATTR),
+	END_ATTRIBUTES_ARRAY,
+};
+
+struct iio_attribute dac_global_attributes[] = {
+	DAC_DEMO_ATTR("dac_global_attr", DAC_GLOBAL_ATTR),
+	END_ATTRIBUTES_ARRAY,
+};
+
+struct iio_channel *iio_dac_demo_channel(uint8_t ch_no)
```
Comment (mchindri) [inline drivers/dac/dac_demo/iio_dac_demo.c:79]: This function is not needed

File: drivers/dac/dac_demo/iio_dac_demo.c [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+	.debug_attributes = NULL,	\
+	.buffer_attributes = NULL,	\
+	.prepare_transfer = update_dac_channels,	\
+	.end_transfer = close_dac_channels,	\
+	.read_dev = (int32_t (*)())dac_write_samples,	\
+	.debug_reg_read = (int32_t (*)()) dac_demo_reg_read,	\
+	.debug_reg_write = (int32_t (*)()) dac_demo_reg_write	\
+}
+
+struct iio_device const dac_demo_iio_descriptor = DAC_DEMO_DEV(
```
Comment (mchindri) [inline drivers/dac/dac_demo/iio_dac_demo.c:130]: Why using the define?

File: drivers/dac/dac_demo/iio_dac_demo.h [hunk 4]
```diff
@@ -41,85 +41,10 @@
 #define IIO_DEMO_DAC
 
 #include <stdlib.h>
+#include <sys/types.h>
```
Comment (mchindri) [inline drivers/dac/dac_demo/iio_dac_demo.h:44]: Is sys/type needed?

File: projects/iio_demo/src/app/main.c [hunk 5]
```diff
@@ -240,12 +242,10 @@ int main(void)
 	status = dac_demo_init(&dac_desc, &dac_init_par);
 	if (status != SUCCESS)
 		return status;
-	ADC_CHANNEL_NO = adc_desc->active_ch;
-	uint32_t adc_channels = ADC_CHANNEL_NO;
-	struct iio_device adc_demo_iio_descriptor = ADC_DEMO_DEV(adc_channels);
-	DAC_CHANNEL_NO = dac_desc->active_ch;
-	uint32_t dac_channels = DAC_CHANNEL_NO;
-	struct iio_device dac_demo_iio_descriptor = DAC_DEMO_DEV(dac_channels);
+//	struct iio_device adc_demo_iio_descriptor =
```
Comment (mchindri) [inline projects/iio_demo/src/app/main.c:245]: Commented lines can be removed

File: drivers/adc/adc_demo/iio_adc_demo.c [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+	.debug_attributes = NULL,	\
+	.buffer_attributes = NULL,	\
+	.prepare_transfer = update_adc_channels,	\
+	.end_transfer = close_adc_channels,	\
+	.read_dev = (int32_t (*)())adc_read_samples,	\
+	.debug_reg_read = (int32_t (*)()) adc_demo_reg_read,	\
+	.debug_reg_write = (int32_t (*)()) adc_demo_reg_write	\
+}
+
+struct iio_device const adc_demo_iio_descriptor = ADC_DEMO_DEV(
```
[the remark this answers is not in the document]
Comment (adrimbarean) [reply drivers/adc/adc_demo/iio_adc_demo.c:113]: The define was here before. Should I remove it?

[the remark this answers is not in the document]
Comment (mchindri) [reply drivers/adc/adc_demo/iio_adc_demo.c:113]: I would say yes

**Comment by adrimbarean** [comment]
V2:
 - remove commented code;
 - remove unwanted defines;
 - remove redundant function.

File: drivers/adc/adc_demo/iio_adc_demo.h [hunk 7]
```diff
@@ -41,85 +41,10 @@
 #define IIO_DEMO_ADC
 
 #include <stdlib.h>
+#include <sys/types.h>
```
[the remark this answers is not in the document]
Comment (mchindri) [reply drivers/adc/adc_demo/iio_adc_demo.h:44]: Strange, it seems it is included already here: https://github.com/analogdevicesinc/no-OS/blob/441ebb5fae69c974f92175ea450d77d724d59aab/libraries/iio/iio_types.h#L48

**Pull request merged**